pep-0677.rst

```diff
@@ -0,0 +1,854 @@
+    introduces no new semantics and closely mimics syntax in other
+    popular languages like Kotlin, Scala, and TypesScript, is much
+    less likely to confuse users.
+
+- We intend to implement the current proposal in a way that is
+  forward-compatible with the more complicated extended syntax. If the
+  community decides after more experience and discussion that we want
+  the additional features, they should be straightforward to propose
+  in the future.
+- We realized that because of overloads, it is not possible to replace
+  all need for Callback Protocols even with an extended syntax. This
+  makes us prefer proposing a simple solution that handles most use
+  cases well.
+
+We confirmed that the current proposal is forward-compatible with
+extended syntax by
+`implementing <https://github.com/stroxler/cpython/tree/callable-type-syntax--extended>`_
+a quick-and-dirty grammar and AST on top of the grammar and AST for
+the current proposal.
+
+
+Syntax Closer to Function Signatures
+------------------------------------
+
+One alternative we had floated was a syntax much more similar to
+function signatures.
+
+In this proposal, the following types would have been equivalent::
+
+    class Function(typing.Protocol):
+        def f(self, x: int, /, y: float, *, z: bool = ..., **kwargs: str) -> bool:
+            ...
+
+    Function = (x: int, /, y: float, *, z: bool = ..., **kwargs: str) -> bool
+
+
+The benefits of this proposal would have included:
+
+- Perfect syntactic consistency between signatures and callable types.
+- Support for more features of function signatures (named, optional,
+  variadic args) that this PEP does not support.
+
+Key downsides that led us to reject the idea include the following:
+
+- A large majority of use cases only use positional-only arguments,
+  and this syntax would be more verbose for that use case, both
+  because of requiring argument names and an explicit ``/``, for
+  example ``(int, /) -> bool`` where our proposal allows ``(int) ->
+  bool``
+- The requirement for explicit ``/`` for positional-only arguments has
+  a high risk of causing frequent bugs - which often would not be
+  detected by unit tests - where library authors would accidentally
+  use types with named arguments.
+- Our analysis suggests that support for ``ParamSpec`` is key, but the
+  scoping rules laid out in PEP 612 would have made this difficult.
+
+
+Other Proposals Considered
+--------------------------
+
+Functions-as-Types
+~~~~~~~~~~~~~~~~~~
+
+An idea we looked at very early on was to `allow using functions as
+types
+<https://docs.google.com/document/d/1rv6CCDnmLIeDrYlXe-QcyT0xNPSYAuO1EBYjU3imU5s/edit?usp=sharing>`_.
+The idea is allowing a function to stand in for its own call
+signature, with roughly the same semantics as the ``__call__`` method
+of Callback Protocols. Think this may be a great idea and worth its
+own PEP, but that it is not a good alternative to improving the
+usability of callable types:
```

Comment:
Quick example might be good (just like the other callback protocol snippets).

Can you make this sentence less informal? "Think this may be a great idea ..."

Reply:
Yeah, I shied away from an example because it looks kinda weird if we snake case (which is always pythonic for functions, never pythonic for types), and it also looks kinda weird if we camel case (which is always pythonic for types, never pythonic for functions). But I guess I should pick one and just look weird.